src/Symfony/Bundle/FrameworkBundle/Test/WebTestCase.php

```diff
@@ -31,6 +31,10 @@ abstract class WebTestCase extends KernelTestCase
      */
     protected static function createClient(array $options = [], array $server = [])
     {
+        if (true === static::$booted) {
```

Comment:
Shouldn't this check be done in `bootKernel()`?

Reply:
I don't think so because in bootKernel the kernel is shutdown before being booted, the errors occurs when booting the kernel manually and then creating the client

Reply:
shouldnt we just "not boot the kernel" if it's already booted and thus reuse it instead? :thinking:

Reply:
I like @ro0NL suggestion.

Reply:
That would break tests that rely on `bootKernel()` having a freshly booted kernel. Thus meaning one would then have to shut it down explicitly before. The only way for an upgrade path for this is that we need to require consumers to explicitly call shutdown first.

Reply:
The thing is if you have two calls of createClient() that means you need to shutdown before. maybe that's not the right way of fixing the tests but that's how I've done it.

Reply:
Maybe we could add a param to say to reuse one if it exist or to destroy and reboot one, by default it could destroy and reboot one, which means that we could not have these kind of edges cases anymore and the behaviour will be the same.

Reply:
Maybe adding a new `getClient()` method that requires an already booted kernel would help here.

Reply:
this has to be done in this PR ?